atc/db/check_factory.go

```diff
@@ -116,21 +125,32 @@ func (c *checkFactory) TryCreateCheck(ctx context.Context, checkable Checkable,
 	}
 
 	checkPlan := checkable.CheckPlan(from, interval, resourceTypes.Filter(checkable), sourceDefaults)
-
 	plan := c.planFactory.NewPlan(checkPlan)
 
-	build, created, err := checkable.CreateBuild(ctx, manuallyTriggered, plan)
-	if err != nil {
-		return nil, false, fmt.Errorf("create build: %w", err)
-	}
+	if toDB {
```

Comment:
Yes, Aidan asked the same question. Using db will ensure manual check can be executed. If manual check also uses in-memory build, if the ATC happens to crash, then the manual check will lost. This might not be a big issue in most of cases, but for resources that has very long interval, e.g. `check_every: 2h`, if a user wants to run a force check, and clicks "check" button on the UI, but nothing happens because a backend ATC happens to crash, user will get confused. This is the reason I keep manual check to still use db.

Reply:
Yes, Aidan asked the same question. Using db will ensure manual check can be executed. If manual check also uses in-memory build, if the ATC happens to crash, then the manual check will lost. This might not be a big issue in most of cases, but for resources that has very long interval, e.g. check_every: 2h, if a user wants to run a force check, and clicks "check" button on the UI, but nothing happens because a backend ATC happens to crash, user will get confused. This is the reason I keep manual check to still use db.

Reply:
I see, I understand where you are coming from. I do have some reservations about introducing this extra complexity just for covering the rare cases of the ATC crashing during a manual check but I appreciate the effort into making the user experience as smooth as possible for every case!